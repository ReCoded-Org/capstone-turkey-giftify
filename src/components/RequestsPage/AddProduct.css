.addProductMainDiv {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  margin: 10% 0%;
}

.addProductMainDiv > form {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-around;
  align-items: center;
  align-content: stretch;
  background-color: #90e0ef;
  width: 65%;
  border-radius: 15px;
}

.addProductMainDiv > form > label {
  display: flex;
  flex-direction: column;
  flex-wrap: nowrap;
  justify-content: space-between;
  align-items: center;
  align-content: stretch;
  width: 65%;
  font-weight: bold;
  font-size: 20px;
}
.addProductMainDiv > form > label > select,
.addProductMainDiv > form > label > input,
.addProductMainDiv > form > label > textarea {
  width: 100%;
}

.addProductMainDiv > form > label > span {
  margin: 3% 0%;
}

.buttonDivAddProduct {
  width: 100%;
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: center;
  align-content: stretch;
  margin: 5% 0%;
}

@media only screen and (min-width: 750px) and (max-width: 1009px) {
  .addProductMainDiv > form > label {
    width: 50%;
    font-weight: bold;
    font-size: 17.5px;
  }

  .addProductMainDiv > form {
    width: 60%;
  }
}

@media only screen and (min-width: 120px) and (max-width: 749px) {
  .addProductMainDiv > form > label {
    width: 80%;
    font-weight: bold;
    font-size: 17.5px;
  }

  .addProductMainDiv > form {
    width: 85%;
  }
}
