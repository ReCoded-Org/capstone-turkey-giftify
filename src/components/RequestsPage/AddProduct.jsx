import React, { useState, useEffect } from "react";
import db from "./../../firebase";
import "./AddProduct.css";
import { Button } from "react-bootstrap";
import { useHistory } from "react-router-dom";

const AddProduct = () => {
  // eslint-disable-next-line
  const [userID, setUserID] = useState("2");
  const [itemName, setItemName] = useState("");
  const [productCategory, setProductCategory] = useState({ value: "books" });
  const [description, setDescription] = useState("");
  const [condition, setCondition] = useState("");
  const [image, setImage] = useState("");
  const [categoriesName, setCategoriesName] = useState([]);
  const history = useHistory();

  const getCategories = async () => {
    const collection = await db.collection("categories").get();
    const arrayOfDocs = collection.docs.map((doc) => {
      return { id: doc.id, ...doc.data() };
    });
    setCategoriesName(arrayOfDocs);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    db.collection("categories")
      .doc(productCategory.value)
      .collection("items")
      .add({
        userID,
        itemName,
        description,
        condition,
        image,
      })
      .then(history.push("/requestsPage"));
  };

  const handleChange = (e) => {
    setProductCategory({ value: e.target.value });
  };

  useEffect(() => {
    getCategories();
    // eslint-disable-next-line
  }, []);

  return (
    <div className="addProductMainDiv">
      <form onSubmit={handleSubmit}>
        <label>
          <span>Product's Image:</span>
          <input
            type="text"
            onChange={(e) => setImage(e.target.value)}
            value={image}
          />
        </label>
        <label>
          <span>Product's Name:</span>
          <input
            type="text"
            onChange={(e) => setItemName(e.target.value)}
            value={itemName}
          />
        </label>
        <label>
          <span>Product's Category:</span>
          <select value={productCategory.value} onChange={handleChange}>
            {categoriesName.map((info) => {
              return (
                <option key={info.id} value={info.id}>
                  {info.id.charAt(0).toUpperCase() + info.id.slice(1)}
                </option>
              );
            })}
          </select>
        </label>
        <label>
          <span>Product's Short Brief:</span>
          <textarea
            type="text"
            onChange={(e) => setDescription(e.target.value)}
            style={{ resize: "vertical" }}
            value={description}
          />
        </label>
        <label>
          <span>Product's Condition:</span>
          <input
            type="text"
            onChange={(e) => setCondition(e.target.value)}
            value={condition}
          />
        </label>
        <div className="buttonDivAddProduct">
          <Button variant="primary" type="submit">
            Add Product
          </Button>
        </div>
      </form>
    </div>
  );
};

export default AddProduct;
