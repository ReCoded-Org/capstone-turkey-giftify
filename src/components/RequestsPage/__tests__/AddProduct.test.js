import React from "react";
import AddProduct from "./../AddProduct";
import { cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";

afterEach(cleanup);

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useParams: jest
    .fn()
    .mockReturnValue({ category: "clothing", id: "3iz39LBgqzm67DAOhPy1" }),
}));

it("AddProduct rendered correctly", () => {
  const tree = renderer.create(<AddProduct />).toJSON();
  expect(tree).toMatchSnapshot();
});
