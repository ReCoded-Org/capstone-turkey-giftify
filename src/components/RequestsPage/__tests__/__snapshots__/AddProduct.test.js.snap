// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddProduct rendered correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <label>
      Product's Name:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Product's Category:
      <select
        onChange={[Function]}
        value="books"
      />
    </label>
    <label>
      Product's Short Brief:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Product's Condition:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <label>
      Product's Image:
      <input
        onChange={[Function]}
        type="text"
        value=""
      />
    </label>
    <button
      className="btn btn-primary"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
</div>
`;
